experimental/juvix/contract.jvx

```diff
@@ -0,0 +1,198 @@
+let Map = Data.Map
+
+-- common module interfaces for data structures in Common?
+sig MonoidMeasure : Common.Measure
+let MonoidMeasure =
+  type T = Nat
+
+  -- (<>) is normal semi-group type class function
+  let (<>)    = (<>)
+  let measure = id
+end
+
+-- need OrdMap, as modules that one goes to can't be modules?
```

Comment:
I don't understand this comment, can you explicate?

Reply:
Map can't be a functor as it's a common path. So we need a function inside of it, let's say `OrdMap`. That generates the module we wish to use

Reply:
Right, OK